app/queries/user_analytics_query.rb

```diff
@@ -0,0 +1,97 @@
+class UserAnalyticsQuery
+  include DayHelper
+  include MonthHelper
+  include PeriodHelper
+
+  attr_reader :current_facility, :days_ago, :months_ago
+
+  def initialize(facility, days_ago: 30, months_ago: 6)
+    @current_facility = facility
+    @days_ago = days_ago
+    @months_ago = months_ago
+
+    @monthly_follow_ups_data =
+      MyFacilities::FollowUpsQuery
+        .new(facilities: current_facility, period: :month, last_n: months_ago)
+        .follow_ups
+  end
+
+  def daily_follow_ups
+    MyFacilities::FollowUpsQuery
+      .new(facilities: current_facility, period: :day, last_n: days_ago)
+      .follow_ups
+      .group(:year, :day)
+      .count
+      .map { |group, count| [doy_to_date(*group), count] }
+      .to_h
+  end
+
+  def daily_registrations
+    group_by_time_range = (days_ago.to_i - 1).days.ago.beginning_of_day..Date.current.end_of_day
+
+    current_facility
```

Comment:
Do we have a `MyFacilities::Registrations` query that can give us this number? Should we use it for consistency?

Reply:
Not quite. The backing view that it uses primarily has this issue:

It counts up (aggregates) the `registration_count` against the day/month/quarter and hence cannot expose the individual `patient_id`. We want a flatter access to the Patient data so that we can group by `gender` later. I considered making that itself a view, but that would be redundant since it'll be the same amount of rows as just making a normal query to the database; potentially even *less* performant that a normal query.